google/cloud/documentai_toolbox/wrappers/document.py

```diff
@@ -771,3 +802,26 @@ def export_hocr_str(self, title: str) -> str:
         template = environment.get_template("hocr_document_template.xml.j2")
         content = template.render(pages=self.pages, title=title)
         return content
+
+    def to_documentai_document(self) -> documentai.Document:
+        r"""Exports a documentai.Document from the wrapped document with shards merged.
+
+        Args:
+            None.
+        Returns:
+            documentai.Document:
+                Document with all shards merged and text offsets applied.
+        """
+        if len(self.shards) == 1:
+            return self.shards[0]
+
+        merged_document = documentai.Document(text=self.text, pages=[], entities=[])
+        for shard in self.shards:
+            _apply_text_offset(
```

Comment:
Since the current implementation of `_apply_test_offset` modifies the shard in-place, after calling `to_documentai_document` the `Document` objects is mutated too.  This side effect is undesirable since we cannot be sure whether the same `Document` object needs to be used for other purposes after the call.

Reply:
Ah, that's correct. I didn't intend to modify everything in place, but my original implementation didn't seem to modify as expected with all of the recursion involved. I'll try making a full copy